### PR TITLE
Adding the rest of string sdk

### PR DESCRIPTION
**Pull request by michelchan**
Adding the following functions:
```
String    append
String    concat
String    contains
String    dropLeft
String    dropRight
String    endsWith
String    join
String    length
String    padLeft
String    padRight
String    right
String    slice
String    split
String    startsWith
String    toFloat
String    toLower
String    toUpper
String    trim
String    trimLeft
String    trimRight
```

### CODE REVIEW

File: examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/StringTests.elm [hunk 1]
```diff
@@ -57,6 +182,30 @@ stringFromFloatTest ctx =
...
+        (String.toFloat "1.5")
+
+
+{-|
+
+    Test: String/toFloat - Invalid
+    expected = Nothing
+
+-}
+stringToFloatTest2 : TestContext -> Maybe Float
```
[the remark this answers is not in the document]
Comment (michelchan) [reply examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/StringTests.elm:203]: Agreed 😆 there were already tests in the file that were named 1 and 2, so i kept with the same convention. I'll change it

File: examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm [hunk 2]
```diff
@@ -1875,42 +1881,49 @@ basicsAlwaysTest x =
     in
     f [ 4 ]
 
+
 {-| Test: SdkBasics/basicsClampTest
 -}
 basicsClampTest : a -> a -> a -> a
```
Comment (edwardpeters) [inline examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm:1887]: I'd change all of these type variables from `a` to `number`... IDK if it's the intention that that be a rigidly enforced in morphir, but it is in elm, and enforced or not is the only way of indicating that the type must belong to the number typeclass.

File: examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm [hunk 3]
```diff
@@ -1875,42 +1881,49 @@ basicsAlwaysTest x =
...
 {-| Test: SdkBasics/basicsIdentityTest
 -}
 basicsIdentityTest : a -> a
 basicsIdentityTest x =
     x
 
+
 {-| Test: SdkBasics/basicsPowerTest
 -}
 basicsPowerTest : a -> a -> a
```
Comment (edwardpeters) [inline examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm:1901]: See above regarding `number`

File: examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm (hunk 2 above)
Comment (michelchan) [reply examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm:1887]: i can do this in a separate pr since this was just part of running the formatter

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    (context: NativeContext) => (ref: RTString, str: RTString) =>
+      val result = str.value.endsWith(ref.value)
+      RT.Primitive.Boolean(result)
+  }
+
+  val join = DynamicNativeFunction2("join") {
+    (context: NativeContext) => (sep: RTString, list: RT.List) =>
+      val result = list.elements.zipWithIndex.foldLeft("") { (x, y) =>
+        val withSep =
+          if (y._2 < list.elements.length - 1) coerceString(y._1).value ++ sep.value else coerceString(y._1).value
```
Comment (edwardpeters) [inline morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:60]: I think this implementation works, but I'd suggest just mapping the elements to a list of strings and then using Scala's `mkString` method rather than rolling your own.

File: examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm (hunk 2 above)
Comment (edwardpeters) [reply examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm:1887]: The formatter changed `number` to `a`?

Comment (edwardpeters) [reply examples/morphir-elm-projects/evaluator-tests/src/Morphir/Examples/App/SdkBasicTests.elm:1887]: Oh wait, nvm I see what you meant, sure

**Review by edwardpeters — APPROVED** [review]

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala (hunk 4 above)
Comment (michelchan) [reply morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:60]: I couldn't figure out how to do this cleanly because in a `fold`, y would be a RTValue. Should I do coerceString(y)?

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+  val slice = DynamicNativeFunction3("slice") {
+    (context: NativeContext) => (start: RT.Primitive.Int, end: RT.Primitive.Int, str: RTString) =>
+      val result = str.value.slice(start.valueAsInt, end.valueAsInt)
+      RTString(result)
+  }
+
+  val split = DynamicNativeFunction2("split") {
+    (context: NativeContext) => (sep: RTString, str: RTString) =>
+      val result = str.value.split(sep.value).toList.map(RTString(_))
```
[the remark this answers is not in the document]
Comment (michelchan) [reply morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:98]: That was an edge case I didn't think about. I'll add some test cases for this too + comments

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala (hunk 4 above)
Comment (michelchan) [reply morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:60]: That works, thanks!

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala (hunk 5 above)
[the remark this answers is not in the document]
Comment (DamianReeves) [reply morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:94]: It this stage in this effort I want to say lets match the semantics present in `finos/morphir-jvm` .

File: morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+  val right = DynamicNativeFunction2("right") {
+    (context: NativeContext) => (n: RT.Primitive.Int, str: RTString) =>
+      val result = str.value.substring(str.value.length - n.valueAsInt)
+      RTString(result)
+  }
+
+  val slice = DynamicNativeFunction3("slice") {
+    (context: NativeContext) => (start: RT.Primitive.Int, end: RT.Primitive.Int, str: RTString) =>
+      val result = str.value.slice(start.valueAsInt, end.valueAsInt)
```
[the remark this answers is not in the document]
Comment (DamianReeves) [reply morphir/src/org/finos/morphir/runtime/sdk/StringSDK.scala:88]: I'm okay with filing an issue (and JIRA) to track the need to align slice functionality... I do think it is desired to have this behavior match
